src/components/Filters/FilterDropdownMenu.test.tsx

```diff
@@ -0,0 +1,179 @@
+import { useState } from "react";
+import { click, render } from "src/utils/rtl";
+import { checkboxFilter } from "./CheckboxFilter";
+import { FilterDropdownMenu } from "./FilterDropdownMenu";
+import { multiFilter } from "./MultiFilter";
+import { singleFilter } from "./SingleFilter";
+import { FilterDefs } from "./types";
+import { GroupByConfig } from "./utils";
+
+describe("FilterDropdownMenu", () => {
+  it("shows filters when button is clicked", async () => {
+    type TestFilter = { status?: string };
+    const r = await render(
+      <TestFilterDropdownMenu<TestFilter>
+        filterDefs={{
+          status: singleFilter({
+            options: [
+              { id: "active", name: "Active" },
+              { id: "inactive", name: "Inactive" },
+            ],
+            getOptionValue: (o) => o.id,
+            getOptionLabel: (o) => o.name,
+          }),
+        }}
+      />,
+    );
+
+    // Filters should not be visible initially
+    expect(r.queryByTestId("filterDropdown_status")).not.toBeInTheDocument();
+
+    // Click the filter button
+    click(r.filterDropdown_button);
+
+    // Filters should now be visible
+    expect(r.filterDropdown_status).toBeInTheDocument();
+  });
+
+  it("displays chips when filters are selected and dropdown is closed", async () => {
+    type TestFilter = { status?: string };
+    const r = await render(
+      <TestFilterDropdownMenu<TestFilter>
+        filterDefs={{
+          status: singleFilter({
+            options: [
+              { id: "active", name: "Active" },
+              { id: "inactive", name: "Inactive" },
+            ],
+            getOptionValue: (o) => o.id,
+            getOptionLabel: (o) => o.name,
+          }),
+        }}
+        initialFilter={{ status: "active" }}
+      />,
+    );
+
+    // Chip should be visible when dropdown is closed
+    expect(r.filterDropdown_chip_status).toBeInTheDocument();
+    expect(r.filterDropdown_chip_status).toHaveTextContent("Active");
+  });
+
+  it("removes filter when chip is clicked", async () => {
+    type TestFilter = { status?: string };
+    const r = await render(
+      <TestFilterDropdownMenu<TestFilter>
+        filterDefs={{
+          status: singleFilter({
+            options: [
+              { id: "active", name: "Active" },
+              { id: "inactive", name: "Inactive" },
+            ],
+            getOptionValue: (o) => o.id,
+            getOptionLabel: (o) => o.name,
+          }),
+        }}
+        initialFilter={{ status: "active" }}
+      />,
+    );
+
+    // Click the chip
+    click(r.filterDropdown_chip_status);
+
+    // Filter should be cleared
+    expect(r.filterValue).toHaveTextContent("{}");
+  });
+
+  it("shows all chips with wrapping", async () => {
```

Comment:
What is the wrapping part of this test? Not sure we can tell/know/or care whether the fake jsdom is wrapping these or not?